test_scripts/API/MainMenuUpdating/003_on_update_sub_menu_success.lua

```diff
@@ -0,0 +1,85 @@
+---------------------------------------------------------------------------------------------------
+-- HMI requests a subemenu is populated
+
+-- Pre-conditions:
+-- a. HMI and SDL are started
+-- b. appID is registered and activated on SDL
+-- c. mobile sends an addSubMenu request with no other contents
+
+-- Steps:
+-- User opens the menu, and the hmi sends UI.OnUpdateSubMenu
+
+-- Expected:
+-- Mobile receives notification that the submenu should be updated
+---------------------------------------------------------------------------------------------------
+
+--[[ Required Shared libraries ]]
+local runner = require('user_modules/script_runner')
+local common = require('test_scripts/Smoke/commonSmoke')
+
+--[[ Test Configuration ]]
+runner.testSettings.isSelfIncluded = false
+
+local addSubMenu = {
+    menuID = 50,
+    menuName = "Sub Menu"
+}
+
+local addSubMenuHMI = {
+    menuID = 50,
+    menuParams = {
+        menuName = "Sub Menu"
+    }
+}
+
+local onSystemContextParams = {
+    systemContext = "MENU"
+}
+
+local onUpdateSubMenu = {
```

Comment:
i like how you added `appID = nil` on the other notification that it is added dynamically to